conda-store-server/conda_store_server/server/auth.py

```diff
@@ -233,31 +317,21 @@ def is_subset_entity_permissions(self, entity, new_entity):
                 return False
         return True
 
-    def authorize(self, entity, arn, required_permissions):
+    def authorize(
+        self, entity: schema.AuthenticationToken, arn: str, required_permissions
+    ):
         return required_permissions <= self.get_entity_permissions(
             entity=entity, arn=arn
         )
 
-    def database_role_bindings(self, entity):
-        with self.authentication_db() as db:
-            result = db.execute(
-                text(
-                    """
-                    SELECT nrm.entity, nrm.role
-                    FROM namespace n
-                    RIGHT JOIN namespace_role_mapping nrm ON nrm.namespace_id = n.id
-                    WHERE n.name = :primary_namespace
-                    """
-                ),
-                {"primary_namespace": entity.primary_namespace},
-            )
-            raw_role_mappings = result.mappings().all()
-
-            db_role_mappings = defaultdict(set)
-            for row in raw_role_mappings:
-                db_role_mappings[row["entity"]].add(row["role"])
-
-        return db_role_mappings
+    def database_role_bindings(self, entity: schema.AuthenticationToken):
+        # This method can be reached from the router_ui via filter_environments.
+        # Since the UI routes are not versioned, we don't know which API version
```

Comment:
Why are ui routes not versioned? should they be?

Reply:
This is out of scope of this PR. Maybe. It's just how `router_ui` is defined.

But even if these were versioned, it wouldn't solve the problem here because nothing prevents someone from calling HTTP routes that access v1 and v2 tables. If they do, which table should be used here? Well, no way to tell without the command-line switch, which is why I introduced it, because it shows the intent and allows to ignore the other table.

Reply:
I agree it is out of scope, but I think it is worth talking about. Maybe not in this PR, but we should probably add a new issue to discuss it.